android/filament-android/src/main/java/com/google/android/filament/Texture.java

```diff
@@ -29,6 +29,33 @@
 
 import static com.google.android.filament.Texture.Type.COMPRESSED;
 
+/**
+ * Texture
+ * <p>The <code>Texture</code> class supports:</p>
+ * <ul>
+ *  <li>2D textures</li>
+ *  <li>3D textures</li>
+ *  <li>Cube maps</li>
+ *  <li>mip mapping</li>
+ * </ul>
+ *
+ *
+ * <h1>Creation and destruction</h1>
+ *
+ * A <code>Texture</code> object is created using the {@link Texture.Builder} and destroyed by
+ * calling {@link Engine#destroyTexture}.
+ *
+ * <pre>
+ *  Engine engine = Engine.create();
+ *
+ *  Texture* texture = new Texture.Builder()
```

Comment:
1) Java doesn't have pointers, I think.  :)

2) I've been preceding my usage examples with `Kotlin usage example:`, I think there's not much difference for Kotlin vs Java in these simple cases, just remove the semicolons.  I can go either way but we should probably be consistent.  (I've heard that 50% of the top 1k apps in the Play Store are Kotlin.)

3) It might be helpful to add an abbreviated codeline to your example for the upload, something like: `texture.setImage(...)`

4) I think it would be helpful to mention in the doc that Textures are bound via `MaterialInstance#setParameter`

5) Add @see PixelBufferDescriptor

Reply:
yeah I missed several pointers before. 
I think snippet examples in the javadoc should be in the java language -- I see it as the more "universal" language, like it's more "bare". It's also known by more people, taught in schools, etc..

Reply:
Sounds good, I'll switch my Kotlin snippets to Java.